infra/lambda-stub.tf

```diff
@@ -0,0 +1,203 @@
+# stubs/authorizer.js
+
+resource "aws_lambda_function" "stubs_authorizer" {
+  function_name    = "identity-stubs-authorizer-${terraform.workspace}"
+  filename         = "${path.module}/../lambda/src/stubs/authorizer.js.zip"
+  handler          = "authorizer.handler"
+  role             = aws_iam_role.identity_api_gateway_lambda_role.arn
+  runtime          = "nodejs12.x"
+  source_code_hash = filebase64sha256("${path.module}/../lambda/src/stubs/authorizer.js.zip")
+
+  depends_on = [data.archive_file.stubs_authorizer_zip]
+
+  tags = merge(
+    local.common_tags,
+    map(
+      "name", "identity-stubs-authorizer-${terraform.workspace}"
+    )
+  )
+}
+
+data "archive_file" "stubs_authorizer_zip" {
+  type        = "zip"
+  source_file = "${path.module}/../lambda/src/stubs/authorizer.js"
+  output_path = "${path.module}/../lambda/src/stubs/authorizer.js.zip"
+}
+
+resource "aws_lambda_permission" "stubs_authorizer" {
+  statement_id  = "AllowAPIGatewayInvoke"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.stubs_authorizer.function_name
+  principal     = "apigateway.amazonaws.com"
+  source_arn    = "${aws_api_gateway_rest_api.identity.execution_arn}/*/*"
+}
+
+# stubs/empty-200.js
+
+resource "aws_lambda_function" "stubs_empty-200" {
+  function_name    = "identity-stubs-empty-200-${terraform.workspace}"
+  filename         = "${path.module}/../lambda/src/stubs/empty-200.js.zip"
```

Comment:
The pattern I've seen for deploying Lambdas across existing Wellcome infrastructure is to store the Lambda zip in an S3 bucket and configure the function to point to that instead of a local file (see https://github.com/wellcomecollection/goobi-infrastructure/blob/588dad11bf24693aea02c0fc786c01b09dfda0de/infrastructure/staging/lambdas.tf, https://github.com/wellcomecollection/search-logger/blob/a3238dd3195f9424c8b75e73052ba93a211621af/terraform/lambda.tf).

I think this is the approach we should be going for. It keeps the build process decoupled from the deployment process in the same way that producing the Docker image as a build artifact and consuming it later in the deployment would be. Right now you will always need to re-build the Lambdas with the currently deployed version of the source code to avoid clobbering the previously deployed Lambda.

Not a problem in itself, but can be a bit of a nuisance. As a bonus, rollbacks also become possible without rebuilding the commit you want to rollback to if we key those zip files with a version number (e.g. users-list-v0.1.0.js.zip).

This might be better placed for when we have the TypeScript project and build in place (see https://github.com/wellcomecollection/search-logger/blob/d191e397c80814d909cbefd11971081402485980/lambda/package.json#L7 for an example of that), so I could pick it up as part of my work, but interested on your thoughts on this.

Reply:
I don't disagree and happy for this to be picked up when we start delivering the Lambda's properly - for the purposes of these stubs, we can stick with the noddy approach in this PR.

Reply:
Nice spot @garyttierney.